.github/workflows/build.yml

```diff
@@ -461,44 +461,77 @@ jobs:
+      - name: Prepare Chart Release and index entry
         if: ${{ needs.setup.outputs.run_build == 'true' }}
         env:
           BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
-          GITHUB_REF: ${{ github.ref }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           REPORT_CONTENT: ${{ needs.chart-verifier.outputs.report_content }}
-          CHART_ENTRY_NAME: ${{ needs.chart-verifier.outputs.chart_entry_name }}
           REDHAT_TO_COMMUNITY: ${{ needs.chart-verifier.outputs.redhat_to_community }}
           WEB_CATALOG_ONLY: ${{ needs.chart-verifier.outputs.web_catalog_only }}
           OCP_VERSION_RANGE: ${{ steps.get-ocp-range.outputs.ocp-version-range }}
-        id: release-charts
+        id: prepare-chart-release
         run: |
-          tar zxvf ./scripts/dependencies/helm-chart-releaser/chart-releaser_1.2.0_linux_amd64.tar.gz
-          sudo cp -f cr /usr/local/bin/cr
-          INDEX_BRANCH=$(if [ "${GITHUB_REF}" = "refs/heads/main" ]; then echo "refs/heads/gh-pages"; else echo "${GITHUB_REF}-gh-pages"; fi)
-          CWD=`pwd`
           cd pr-branch
           ../ve1/bin/chart-repo-manager \
             --repository=${{ github.repository }} \
-            --index-branch=${INDEX_BRANCH} \
             --api-url=${{ github.event.pull_request._links.self.href }} \
-            --pr-number=${{ github.event.number }}
-          cd ${CWD}
 
-      - name: Release
+      # Upload the report file, potentially paired with a public key and, if provided, the chart's tarball and its prov file.
+      # Only the report file is always included.
+      # The release tag format is <organization_name>-<chart_name>-<chart_version>
+      - name: Create GitHub release
         if: ${{ needs.chart-verifier.outputs.web_catalog_only == 'False' }}
-        uses: softprops/action-gh-release@v0.1.12
-        continue-on-error: true
+        uses: softprops/action-gh-release@v0.1.15
         with:
           tag_name: ${{ needs.chart-verifier.outputs.release_tag }}
           files: |
-              ${{ steps.release-charts.outputs.report_file }}
-              ${{ steps.release-charts.outputs.public_key_file }}
+              ${{ steps.prepare-chart-release.outputs.report_file }}
+              ${{ steps.prepare-chart-release.outputs.public_key_file }}
+              ${{ steps.prepare-chart-release.outputs.path_to_chart_tarball }}
+              ${{ steps.prepare-chart-release.outputs.prov_file_name }}
           fail_on_unmatched_files: true
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Update Helm repository index
+        if: ${{ needs.setup.outputs.run_build == 'true' }}
+        env:
+          CHART_ENTRY_NAME: ${{ needs.chart-verifier.outputs.chart_entry_name }}
+          WEB_CATALOG_ONLY: ${{ needs.chart-verifier.outputs.web_catalog_only }}
+        run: |
+          INDEX_BRANCH=$(if [ "${GITHUB_REF}" = "refs/heads/main" ]; then echo "gh-pages"; else echo "${GITHUB_REF##*/}-gh-pages"; fi)
```

Comment:
Just a note: this seems like one of those metadata items that we can derive and set workflow-wide.
No action required for this comment. Just wanted to mention it so it's on your radar for the issue on the same topic. (Feel free to "Resolve Conversation" once read).

Reply:
Good idea, I'll keep this in mind for a future iteration on this.